src/main/java/org/opensrp/web/rest/EventResource.java

```diff
@@ -283,15 +290,31 @@ private EventSyncBean getEventsAndClients(EventSearchBean eventSearchBean, Integ
 				clients.addAll(clientService.findByFieldValue(BASE_ENTITY_ID, clientIds.subList(i, end)));
 			}
 			logger.info("fetching clients took: " + (System.currentTimeMillis() - startTime));
-
+			
 			searchMissingClients(clientIds, clients, startTime);
 			
 			if (returnCount) {
 				totalRecords = eventService.countEvents(eventSearchBean);
 			}
 			
 		}
-
+		
+		//PII Data masking 
+		//TO DO research on ways to improve this
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		String currentPrincipalName = authentication.getName();
+		User user = openmrsUserService.getUser(currentPrincipalName);

```

Comment:
Use this to get logged in user `org.opensrp.web.rest.RestUtils.currentUser(Authentication)`